src/sql/src/func.rs

```diff
@@ -254,22 +252,30 @@ impl<'a> ArgImplementationMatcher<'a> {
             .collect();
         let mut m = Self { ident, ecx, impls };
 
-        let mut raw_arg_types = Vec::new();
+        let mut exprs = Vec::new();
         for arg in args {
-            let expr = super::query::plan_expr(ecx, &arg, Some(ScalarType::Unknown))?;
-            raw_arg_types.push(ecx.scalar_type(&expr));
-            Self::clear_expr_param_data(ecx, &arg);
+            let expr = super::query::plan_coercible_expr(ecx, arg)?.0;
+            exprs.push(expr);
         }
 
-        // Exact match
-        let f = if let Some(func) = m.get_implementation(&raw_arg_types) {
-            func
-        } else {
-            // Best match
-            m.best_match(args, &raw_arg_types)?
-        };
+        let f = m.best_match(&exprs)?;
```

Comment:
without an attempt at an exact match, `best_match` seems misnamed

Reply:
Shoot, I forgot I removed that. I guess it's not strictly necessary given that everything passed without it, but I added it back so that we could follow Postgres's algorithm exactly.